plugins/entity-feedback/src/components/FeedbackResponseDialog/FeedbackResponseDialog.tsx

```diff
@@ -77,18 +80,54 @@ export const FeedbackResponseDialog = (props: FeedbackResponseDialogProps) => {
     feedbackDialogTitle = 'Please provide feedback on what can be improved',
     open,
     onClose,
+    enableValidation = false,
+    showCommentsTextBox = true,
   } = props;
   const classes = useStyles();
   const errorApi = useApi(errorApiRef);
   const feedbackApi = useApi(entityFeedbackApiRef);
-  const [responseSelections, setResponseSelections] = useState(
-    Object.fromEntries(feedbackDialogResponses.map(r => [r.id, false])),
-  );
+  const [responseSelections, setResponseSelections] = useState(() => {
+    const initialSelections = Object.fromEntries(
+      feedbackDialogResponses.map(r => [r.id, false]),
+    );
+
+    if (
+      feedbackDialogResponses.some(response => response.id === 'other') &&
+      !initialSelections.hasOwnProperty('other')
+    ) {
+      initialSelections.other = false;
+    }
+
+    return initialSelections;
+  });
   const [comments, setComments] = useState('');
   const [consent, setConsent] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
 
   const [{ loading: saving }, saveResponse] = useAsyncFn(async () => {
     try {
+      setErrorMessage('');
+
+      if (
+        enableValidation &&
+        Object.keys(responseSelections).every(key =>
+          key !== 'other' ? !responseSelections[key] : true,
+        )
+      ) {
+        setErrorMessage('Please select at least one reason.');
+        return;
+      }
+
+      if (
+        showCommentsTextBox &&
+        (responseSelections.other ||
+          !feedbackDialogResponses.some(response => response.id === 'other')) &&
+        !comments
```

Comment:
I still don't agree with validating comments only if the `other` option is selected - this ends up with inconsistent behavior dependent on whether `other` is an available option. If there is no `other` option then comments are required but if `other` is an option then comments are only required if `other` is chosen. 

I suggest a more generic way to implement this would be to support a way to customize what validation you want to enforce that aligns with what `feedbackDialogResponses` you supply - this could done by providing a validation callback to this component which is called with the current selections/comment content and returns either valid or some error message